pkg/karmadactl/cmdinit/utils/util.go

```diff
@@ -18,13 +18,15 @@ type Downloader struct {
 	Current int64
 }
 
+var output io.Writer = os.Stdout
+
 // Read Implementation of Downloader
 func (d *Downloader) Read(p []byte) (n int, err error) {
 	n, err = d.Reader.Read(p)
 	d.Current += int64(n)
-	fmt.Printf("\rDownloading...[ %.2f%% ]", float64(d.Current*10000/d.Total)/100)
+	fmt.Fprintf(output, "\rDownloading...[ %.2f%% ]", float64(d.Current*10000/d.Total)/100)
 	if d.Current == d.Total {
-		fmt.Printf("\nDownload complete.")
```

Comment:
What if we just add a `\n` at the end? I'm curious about what would happen.

Reply:
I also think it need a `\n` at the end. @prodanlabs  may explain it.

Reply:
This is probably the minimum fix.

The `fsprint` is pretty annoying because some static check tools would raise a warning about unchecked return value...

Reply:
First, we can't append `\n` to `Downloading...[ %.2f%% ]`, otherwise it will print
```
Downloading...[ 1.00% ]
Downloading...[ 2.00% ]
Downloading...[ 3.00% ]
...
```

While we expect these logs are print in one line. e.g:
```
Downloading...[ 100.00% ]
```

Then, `\nDownload complete.` I think we shall append to it. Leave it to another PR after @prodanlabs affirm.

Reply:
Will you accept it, @RainbowMango ?

Reply:
I'll get back to you after I try it.

Reply:
I tried this, and looks fine:
```golang
// Read Implementation of Downloader
func (d *Downloader) Read(p []byte) (n int, err error) {
	n, err = d.Reader.Read(p)
	if err != nil {
		if err == io.EOF {
			return
		}
	}

	d.Current += int64(n)
	fmt.Printf("\rDownloading...[ %.2f%% ]", float64(d.Current*10000/d.Total)/100)
	if d.Current == d.Total {
		println("Download complete.")
	}
	return
}
```

The output:
```bash
Downloading...[ 100.00% ]Download complete.
```

Reply:
```go
		if err == io.EOF {  // mistake? shall be err != io.EOF
			return
		}
```

Reply:
```go
	fmt.Printf("\rDownloading...[ %.2f%% ]", float64(d.Current*10000/d.Total)/100)
	if d.Current == d.Total {
		fmt.Println("\nDownload complete.")
	}
```

Output:
```
Downloading...[ 100.00% ]
Download complete.
```

Test in Goland, and work well.

Reply:
> // mistake? shall be err != io.EOF

Reply:
It's strange. When I use `err == io.EOF`, testing will not print the download info.

```
=== RUN   TestDownloadFile
--- PASS: TestDownloadFile (0.01s)
PASS
```

After change to `err != io.EOF`, download info appears:

```
=== RUN   TestDownloadFile
Downloading...[ 100.00% ]
Download complete.
--- PASS: TestDownloadFile (0.01s)
PASS
```

Reply:
I change it to
https://github.com/karmada-io/karmada/blob/7315de1c5e890525bf330fc0e2f8d68346a45542/pkg/karmadactl/cmdinit/utils/util.go#L22-L36

karmadactl init output:
```
I1027 19:08:36.240173   14705 deploy.go:270] download crds file:https://github.com/karmada-io/karmada/releases/download/v1.3.0/crds.tar.gz
Download complete.     
```

go test output:
```
=== RUN   TestDownloadFile
Download complete.       
--- PASS: TestDownloadFile (0.02s)
PASS
```